pkg/util/pki/asn1_util.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
```

Comment:
suggestion: This file seemingly includes some copy+pasted code from Go, which is fine but it's worth being explicit about it!

Ideally, any copied functions would be in their separate own file so it's crystal clear which parts are copyrighted by the Go team and which parts aren't.

```suggestion

// This file contains some code copied from the Go standard library under the following license: https://github.com/golang/go/blob/c95fe91d0715dc0a8d55ac80a80f383c3635548b/LICENSE
```